jboss-cert-ldap-login-module/src/main/java/org/esbtools/auth/jboss/CertLdapLoginModule.java

```diff
@@ -160,13 +152,13 @@ public void initializeRolesProvider() throws Exception {
             LOGGER.debug("Certificate principal:" + certPrincipal);
 
             //first try getting search name from uid in certificate principle (new certificates)
-            String searchName = getLDAPAttribute(certPrincipal, UID);
+            String searchName = environment.getLDAPAttribute(certPrincipal, UID);
             if(StringUtils.isNotBlank(searchName)) {
                 //only try to validate environment if it is a certificate that contains uid
-                validateEnvironment(certPrincipal);
+                environment.validate(certPrincipal);
             } else {
                 // fallback to getting search name from cn in certificate principle (legacy certificates)
-                searchName = getLDAPAttribute(certPrincipal, CN);
+                searchName = environment.getLDAPAttribute(certPrincipal, CN);
```

Comment:
I think we can actually remove this whole if/else since we don't need to support legacy certificates anymore.  So just:

String searchName = environment.getLDAPAttribute(certPrincipal, UID);
environment.validate(certPrincipal);